README.md

```diff
@@ -1,602 +1,387 @@
+[`route_to`]: https://relishapp.com/rspec/rspec-rails/docs/routing-specs/route-to-matcher
+[`be_routable`]: https://relishapp.com/rspec/rspec-rails/docs/routing-specs/be-routable-matcher
+[`have_http_status`]: https://relishapp.com/rspec/rspec-rails/docs/matchers/have-http-status-matcher
+[`match_array`]: https://relishapp.com/rspec/rspec-rails/docs/matchers/activerecord-relation-match-array
+[`have_been_enqueued`]: https://relishapp.com/rspec/rspec-rails/docs/matchers/have-been-enqueued-matcher
+[`have_enqueued_job`]: https://relishapp.com/rspec/rspec-rails/docs/matchers/have-enqueued-job-matcher
 
-### `assign(key, val)`
+### What else does RSpec Rails add?
 
-Use this to assign values to instance variables in the view:
+For a comprehensive look at RSpec Rails’ features,
+read the [official Cucumber documentation][].
 
-```ruby
-assign(:widget, Widget.new)
-render
-```
+[official Cucumber documentation]: https://relishapp.com/rspec/rspec-rails/docs
 
-The code above assigns `Widget.new` to the `@widget` variable in the view, and
-then renders the view.
+## What tests should I write?
 
-Note that because view specs mix in `ActionView::TestCase` behavior, any
-instance variables you set will be transparently propagated into your views
-(similar to how instance variables you set in controller actions are made
-available in views). For example:
+At the risk of grossly oversimplifying,
+there are two kinds of tests you can write:
+those that verify
+**an individual component of your application** (unit tests),
+and those that verify
+**how multiple parts of your application work as a whole** (integration tests).
 
-```ruby
-@widget = Widget.new
-render # @widget is available inside the view
-```
+RSpec Rails defines ten different _types_ of specs (7 unit, 3 integration)
+for testing different parts of a typical Rails application.
+Many inherit from one of Rails’ built-in `TestCase` classes,
+meaning the helper methods provided by default in Rails tests
+are available in RSpec, as well.
 
-RSpec doesn't officially support this pattern, which only works as a
-side-effect of the inclusion of `ActionView::TestCase`. Be aware that it may be
-made unavailable in the future.
+- **Unit**
 
-#### Upgrade note
+  | Spec type        | Corresponding Rails test class     |
+  | ---------------- | ---------------------------------- |
+  | [model][]        |                                    |
+  | [controller][]\* | [`ActionController::TestCase`][]   |
+  | [mailer][]       | `ActionMailer::TestCase`           |
+  | [job][]          |                                    |
+  | [view][]         | `ActionView::TestCase`             |
+  | [routing][]      |                                    |
+  | [helper][]       | `ActionView::TestCase`             |
+
+- **Integration**
+
+  | Spec type       | Corresponding Rails test class        |
+  | --------------- | ------------------------------------- |
+  | [request][]     | [`ActionDispatch::IntegrationTest`][] |
```

Comment:
These are almost identical to controller tests and replace them (see earlier point about why referring to things as unit tests is unhelpful)

Reply:
Good point; I've removed the unit/integration dichotomy from the document entirely.